hyperparameter_tuning/tensorflow2_mnist/hpo_tensorflow2_mnist.ipynb

```diff
@@ -0,0 +1,370 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prefix = 'mnist'\n",
+    "bucket = sess.default_bucket()\n",
+    "loc = sess.upload_data(path=local_data_dir, bucket=bucket, key_prefix=prefix)\n",
+    "\n",
+    "channels = {\n",
+    "    \"training\": loc,\n",
+    "    \"testing\": loc\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set Up a Hyperparameter Tuning Job\n",
+    "Now we have set up the training job and the input data channels, we are ready to train the model with hyperparameter search.\n",
+    "\n",
+    "We will set up the hyperparamter tuning job via the following steps:\n",
+    "* Define the ranges of hyperparameters we plan to tune, in this example, we are tuning learning rate\n",
+    "* Define the objective metric for the tuning job to optimize\n",
+    "* Create a hyperparameter tuner with above setting, as well as tuning resource configurations \n",
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "For a typical ML model, there are three kinds of hyperparamters:\n",
+    "\n",
+    "- Categorical parameters need to take one value from a discrete set.  We define this by passing the list of possible values to `CategoricalParameter(list)`\n",
+    "- Continuous parameters can take any real number value between the minimum and maximum value, defined by `ContinuousParameter(min, max)`\n",
+    "- Integer parameters can take any integer value between the minimum and maximum value, defined by `IntegerParameter(min, max)`\n",
+    "\n",
+    "Learning rate is a continuous variable, therefore, we define its range\n",
+    "by `ContinuousParameter`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.tuner import ContinuousParameter, HyperparameterTuner\n",
+    "\n",
+    "hyperparamter_range = {\n",
+    "    'learning-rate': ContinuousParameter(1e-4, 1e-3)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next we'll specify the objective metric that we'd like to tune and its definition, which includes the regular expression (Regex) needed to extract that metric from the CloudWatch logs of the training job. In this particular case, our script emits average loss value and we will use it as the objective metric, we also set the `objective_type` to be `Minimize`, so that hyperparameter tuning seeks to minize the objective metric when searching for the best hyperparameter setting."
```

Comment:
```suggestion
    "Next you'll specify the objective metric that we'd like to tune and its definition, which includes the regular expression (Regex) needed to extract that metric from the CloudWatch logs of the training job. In this particular case, your script emits average loss value and you will use it as the objective metric. You also set the `objective_type` to be `Minimize`, so that hyperparameter tuning seeks to minimize the objective metric when searching for the best hyperparameter setting."
```